tests/unit/alloc_test.cc

```diff
@@ -155,6 +157,59 @@ SEASTAR_TEST_CASE(test_memory_diagnostics) {
     return make_ready_future<>();
 }
 
+SEASTAR_THREAD_TEST_CASE(test_cross_thread_realloc) {
```

Comment:
the purpose of this test is to verify the behavior of seastar allocator, not the one provided by libc, so probably we should guard it like
```c++
#ifndef SEASTAR_DEFAULT_ALLOCATOR

SEASTAR_THREAD_TEST_CASE(test_cross_thread_realloc) {
   // ...
}
#endif
```

i.e., to move the `#ifndef SEASTAR_DEFAULT_ALLOCATOR` at the end of this test up before this test.

Reply:
I deliberately left in both modes, because it works in both modes. What's the harm? Leaving it in both modes has some advantages one major one which is:

 - the test will run in ASAN etc which is only available in the non-ss-allocator modes, to catch bugs in it
 
 I have been mostly under the assumption that we should gate tests with `SEASTAR_DEFAULT_ALLOCATOR` only if they would fail or fail to compile in the default allocator mode.

Reply:
I see. Makes sense. As we also have tests testing stand library when testing sstring for the similar purpose. So I don't insist.